### PR TITLE
Added LDAP test for Superset

### PR DESCRIPTION
**Pull request by sbernauer**
## Review Checklist
- [x] Code contains useful comments
- [x] Changelog updated (or not applicable)

### CODE REVIEW

**Comment by siegfriedweber** [comment]
The Superset integration tests do not run out of the box on my machine.

I created a test cluster with the following command:
```
./create_test_cluster.py --kind --operator superset=0.4.0-pr145
```

The first test run failed :

```
ldap-1.4.1/1-install-openldap | test step failed 1-install-openldap
```

Perhaps the timeouts must be increased.

The second test run failed:

```
retrieving API resource for secrets.stackable.tech/v1alpha1, Kind=SecretClass failed: the server could not find the requested resource
```

The `create_test_cluster.py` script was not adapted to install the `secret-operator` so I did this manually:

```
helm install secret-operator stackable-dev/secret-operator --version=0.3.0-nightly
```

The third test run failed:

```
    ldap-1.4.1/10-install-superset | test step failed 10-install-superset
    statefulsets.apps "superset-with-ldap-node-default" not found
```

The RBAC rules were not adapted:

```
2022-03-11T15:17:53.305704Z ERROR stackable_operator::logging::controller: Failed to reconcile object controller.name="supersetclusters.superset.stackable.tech" error=reconciler for object SupersetCluster.v1alpha1.superset.stackable.tech/superset-with-ldap.kuttl-test-cuddly-bluebird failed error.sources=[failed to retrieve authentication class myldap, Kubernetes reported error: ApiError: authenticationclasses.authentication.stackable.tech "myldap" is forbidden: User "system:serviceaccount:default:superset-operator-serviceaccount" cannot get resource "authenticationclasses" in API group "authentication.stackable.tech" at the cluster scope: Forbidden (ErrorResponse { status: "Failure", message: "authenticationclasses.authentication.stackable.tech \"myldap\" is forbidden: User \"system:serviceaccount:default:superset-operator-serviceaccount\" cannot get resource \"authenticationclasses\" in API group \"authentication.stackable.tech\" at the cluster scope", reason: "Forbidden", code: 403 }), ApiError: authenticationclasses.authentication.stackable.tech "myldap" is forbidden: User "system:serviceaccount:default:superset-operator-serviceaccount" cannot get resource "authenticationclasses" in API group "authentication.stackable.tech" at the cluster scope: Forbidden (ErrorResponse { status: "Failure", message: "authenticationclasses.authentication.stackable.tech \"myldap\" is forbidden: User \"system:serviceaccount:default:superset-operator-serviceaccount\" cannot get resource \"authenticationclasses\" in API group \"authentication.stackable.tech\" at the cluster scope", reason: "Forbidden", code: 403 }), authenticationclasses.authentication.stackable.tech "myldap" is forbidden: User "system:serviceaccount:default:superset-operator-serviceaccount" cannot get resource "authenticationclasses" in API group "authentication.stackable.tech" at the cluster scope: Forbidden]
```

So I removed the deployed `superset-operator` and started it outside of the cluster.

The next test run failed:

```
    ldap-1.4.1/11-login | running command: [sh -c kubectl exec -n $NAMESPACE python-0 -- python /tmp/login.py]
    ldap-1.4.1/11-login | 2022-03-11 15:28:12,216 DEBUG: Starting new HTTP connection (1): superset-with-ldap-node-default:8088
    ldap-1.4.1/11-login | 2022-03-11 15:28:12,221 DEBUG: http://superset-with-ldap-node-default:8088 "POST /api/v1/security/login HTTP/1.1" 401 29
    ldap-1.4.1/11-login | command terminated with exit code 1
```

I do not know what went wrong this time.

**Comment by sbernauer** [comment]
Thanks for your feedack @siegfriedweber!
All problems should be resolved by now. Working with `./create_test_cluster.py --kind --operator superset=0.4.0-pr145`

**Comment by sbernauer** [comment]
Thanks for your feedack @siegfriedweber!
All problems should be resolved by now. Working with `./create_test_cluster.py --kind --operator superset=0.4.0-pr145`

File: superset-operator/.ci/integration-tests/hcloud-centos-8/cluster.yaml [hunk 1]
```diff
@@ -13,6 +13,7 @@ spec:
   wireguard: false
   versions:
     _-operator: NIGHTLY
+    commons-operator: NIGHTLY # TODO put in t2
```
Comment (siegfriedweber) [inline superset-operator/.ci/integration-tests/hcloud-centos-8/cluster.yaml:16]: TODO is still open.

File: superset-operator/tests/kuttl/ldap-1.4.1/02-install-test-container.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+metadata:
+  name: python
+  labels:
+    app: python
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: python
+  serviceName: python
```
Comment (siegfriedweber) [inline superset-operator/tests/kuttl/ldap-1.4.1/02-install-test-container.yaml:13]: Why is the service name necessary?

File: superset-operator/tests/kuttl/ldap-1.4.1/20-install-superset-insecure-tls.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    ldap:
+      hostname: will.be.replaced.by.kubectl.patch # We need to use kubeclt patch so that we can dynamically put in the test namespace
+      port: 1636
+      searchBase: ou=users,dc=example,dc=org
+      ldapFieldNames:
+        uid: uid
+        group: memberof
+        givenName: givenName
+        surname: sn
+        email: mail
```
Comment (siegfriedweber) [inline superset-operator/tests/kuttl/ldap-1.4.1/20-install-superset-insecure-tls.yaml:31]: Is there a reason why these field names are set in this Superset instance but not in the others?

File: superset-operator/tests/kuttl/ldap-1.4.1/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+certifi==2021.10.8
+charset-normalizer==2.0.10
+idna==3.3
+requests==2.27.1
+urllib3==1.26.8
```
Comment (siegfriedweber) [inline superset-operator/tests/kuttl/ldap-1.4.1/requirements.txt:5]: Is it intended that some of these versions are older than in the smoke tests?

File: superset-operator/tests/kuttl/ldap-1.4.1/login.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    logging.basicConfig(level=log_level, format='%(asctime)s %(levelname)s: %(message)s', stream=sys.stdout)
+
+    superset_insstances = [
+        "superset-with-ldap-no-tls-node-default",
+        "superset-with-ldap-insecure-tls-node-default",
+        "superset-with-ldap-server-veri-tls-node-default",
+        "superset-with-ldap-mutual-veri-tls-node-default",
+    ]
+
+    for superset_insstance in superset_insstances:
```
Comment (siegfriedweber) [inline superset-operator/tests/kuttl/ldap-1.4.1/login.py:18]: Typos

```suggestion
    for superset_instance in superset_instances:
```

**Review by siegfriedweber — CHANGES_REQUESTED** [review]

File: superset-operator/tests/kuttl/ldap-1.4.1/02-install-test-container.yaml (hunk 2 above)
Comment (sbernauer) [reply superset-operator/tests/kuttl/ldap-1.4.1/02-install-test-container.yaml:13]: Well that's the way StatefulSets work...
`kuttl` seems to accept non-valid StatefulSets, but if you try to `kubectl apply` this file the serviceName is needed for a valid sts.

File: superset-operator/tests/kuttl/ldap-1.4.1/20-install-superset-insecure-tls.yaml (hunk 3 above)
Comment (sbernauer) [reply superset-operator/tests/kuttl/ldap-1.4.1/20-install-superset-insecure-tls.yaml:31]: My intention was kind of "increasing the test coverage", so no real reason. Happy to remove them if you'd like

**Review by siegfriedweber — CHANGES_REQUESTED** [review]
I know that this pull request was not adapted to the latest changes in the operator and therefore was not ready to be reviewed. But I needed a working integration test to test the operator and wrote down all changes I made.

**Comment by sbernauer** [comment]
Thanks for the hints!
Updated the tests, so that they work again

File: superset-operator/tests/kuttl/ldap-1.4.1/requirements.txt (hunk 4 above)
Comment (sbernauer) [reply superset-operator/tests/kuttl/ldap-1.4.1/requirements.txt:5]: Absolutely not ^^ I will look into it

Comment (sbernauer) [reply superset-operator/tests/kuttl/ldap-1.4.1/requirements.txt:5]: updated it

**Review by adwk67 — APPROVED** [review]
All tests now run successfully. As discussed with @sbernauer there a few things that will be addressed in https://github.com/stackabletech/superset-operator/issues/195.

**Pull request merged**